Build e3sm failed on Summit with latest CMake build system

This issue is reproducible on Summit with latest E3SM master branch after PR #3043 is merged.

Steps to reproduce:
```
git clone https://github.com/E3SM-Project/E3SM.git
cd E3SM
git submodule update --init
cd cime/scripts
./create_newcase --case FC5AV1C-H01B_ne4_ne4 --compset FC5AV1C-H01B --res ne4_ne4
cd FC5AV1C-H01B_ne4_ne4
./case.setup
./case.build
```
The error information is:
```
-- The C compiler identification is XL 16.1.1.3
...
Found component atm model 'cam'
...
CMake Error at cmake/cmake_util.cmake:53 (message):
  Trying to set flags on non-existent source:
  homme/src/pese/prim_advance_mod.F90, looked for
  components/homme/src/pese/prim_advance_mod.F90
Call Stack (most recent call first):
  cime/scripts/FC5AV1C-H01B_ne4_ne4/Depends.ibm.cmake:40 (e3sm_add_flags)
  cmake/build_model.cmake:150 (include)
  cmake/atm/CMakeLists.txt:1 (build_model)


-- Configuring incomplete, errors occurred!
```

--- a/cime/config/e3sm/machines/Depends.ibm.cmake
+++ b/cime/config/e3sm/machines/Depends.ibm.cmake
@@ -19,8 +19,6 @@ set(PERFOBJS
   homme/src/share/derivative_mod_base.F90
   homme/src/share/bndry_mod_base.F90
   homme/src/theta-l/prim_advance_mod.F90
-  homme/src/pese/prim_advance_mod.F90
-  homme/src/theta/prim_advance_mod.F90
   homme/src/preqx/share/prim_advance_mod.F90
   cam/src/physics/cam/uwshcu.F90
   cam/src/chemistry/aerosol/wetdep.F90)
@@ -49,8 +47,8 @@ endif()
 # These files take long time to compile with default optimization flags.
 # Reducing optimization gives <1min build-times and little impact on model run time.
 # begin
-list(APPEND NOOPT_FILES cam/src/utils/buffer.F90)
+list(APPEND NOOPT_FILES ${CMAKE_CURRENT_BINARY_DIR}/buffer.F90)
 
-foreach(ITEN IN LISTS NOINLINE)
+foreach(ITEM IN LISTS NOINLINE)
   e3sm_add_flags("${ITEM}" "-Q!")
 endforeach()

